### ISSUE TITLE
Socket child can fail to initialize due to className mismatch

### ISSUE THREAD

**Issue by Lange** [nrkno/tv-automation-atem-connection#74]
* **I'm submitting a ...**
[x] bug report
[ ] feature request
[ ] question about the decisions made in the repository
[ ] question about how to use this project

* **Summary**
At [`src/lib/atemSocket.ts#L114`](https://github.com/nrkno/tv-automation-atem-connection/blob/d4f26ee8393c5314ea819d3d935360a4ed85f159/src/lib/atemSocket.ts#L114), the `name` of the `AtemSocketChild` class can sometimes be a mangled/minified string, such as the single letter `c`.

    Later, when [`threadedClass@src/internalApi.ts#L296`](https://github.com/nytamin/threadedClass/blob/ad8309854dd0c7367dc3d3d0d4728d9d448d4169/src/internalApi.ts#L296) loads the same module, it tries to find a class named `c` on the module export, which does not exist, because the actual class name is `AtemSocketChild`. This results in the following error being printed:

    ```
    INIT error {}
    ```

* **Other information** 
This is when running on a Tessel 2 which runs Node v8.11.3, I haven't tested any other platforms or Node versions.

I don't actually know how or where this mangling of the class name occurs, nor why.

**Comment by Lange** [comment]
Ok, so the root cause is that the build process that the Tessel uses does indeed apply a minification step, and I guess `threadedClass` can fall apart when anything that mangles symbol names is used.

**Comment by Julusian** [comment]
Yeah, that sounds like the cause. I'm not sure on how this could be resolved, any minification that mangles exported names might trigger this?
@nytamin any thoughts?

**Comment by Julusian** [comment]
Except, that surely both the caller and module side should be minified to have the same names so it should still line up?
Could you send me an example of what the minified code looks like? (a PM over slack is good)

**Comment by Lange** [comment]
The problem is that the _export_ is still `AtemSocketChild`, but the _class name_ gets mangled. `threadedClass` currently expects that these two things are always identical.

**Comment by Julusian** [comment]
For my memory: the threadedClass PR will help with this issue. There is another issue with the library getting confused after minification whenever we use `.constructor.name`

**Comment by Lange** [comment]
Running my tessel with `--compress=false` seems to at least help somewhat alleviate this issue for now.

**Comment by Julusian** [comment]
This is resolved in `2.0.0-nightly-20200626-195656-eb05a5c.0`.
But there is still some work needed inside this library to remove other useages of `.constructor.name`
